lib/src/rules/unrelated_type_equality_checks.dart

```diff
@@ -132,40 +130,19 @@ class DerivedClass2 extends ClassBase with Mixin {}
 
 ''';
 
-bool _hasNonComparableOperands(TypeSystem typeSystem, BinaryExpression node) {
-  var left = node.leftOperand;
-  var leftType = left.staticType;
-  var right = node.rightOperand;
-  var rightType = right.staticType;
-  if (leftType == null || rightType == null) {
-    return false;
-  }
-  return !left.isNullLiteral &&
-      !right.isNullLiteral &&
-      _nonComparable(typeSystem, leftType, rightType);
-}
-
-bool _isCoreInt(DartType? type) => type != null && type.isDartCoreInt;
-
-bool _isFixNumIntX(DartType type) {
-  // todo(pq): add tests that ensure this predicate works with fixnum >= 1.1.0-dev
-  // See: https://github.com/dart-lang/linter/issues/3868
-  if (type is! InterfaceType) return false;
-  InterfaceElement element = type.element;
-  if (element.name != 'Int32' && element.name != 'Int64') return false;
-  var uri = element.library.source.uri;
-  if (!uri.isScheme('package')) return false;
-  return uri.pathSegments.firstOrNull == 'fixnum';
-}
-
-bool _nonComparable(
-        TypeSystem typeSystem, DartType leftType, DartType? rightType) =>
-    typesAreUnrelated(typeSystem, leftType, rightType) &&
-    !(_isFixNumIntX(leftType) && _isCoreInt(rightType));
-
 class UnrelatedTypeEqualityChecks extends LintRule {
-  static const LintCode code = LintCode('unrelated_type_equality_checks',
-      "Neither type of the operands of '==' is a subtype of the other.",
+  static const LintCode expressionCode = LintCode(
+      'unrelated_type_equality_checks',
+      uniqueName: 'LintCode.unrelated_type_equality_checks_expression',
+      "The right operand type, '{0}', of the equality comparison is not a "
+          "subtype nor a supertype of the left operand type, '{1}'.",
```

Comment:
The message seems a little long and a little formal, so I took a stab at rewriting it. Not sure whether it's better.

The type of the right operand ('{0}') isn't a subtype or a supertype of the left operand ('{1}').